core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -71,9 +86,15 @@
   private static final Logger LOG = LoggerFactory.getLogger(BaseFileSystem.class);
 
   protected final FileSystemContext mFsContext;
+  protected final AlluxioBlockStore mBlockStore;
+
+  private final Object mCloseLock = new Object();
```

Comment:
If this only guards the closing, can we just synchronize on this BaseFileSystem?

Reply:
Already addressed this. just haven't pushed the commit yet.